src/control/cmd/dmg/storage.go

```diff
@@ -116,54 +119,52 @@ func (cmd *storageScanCmd) Execute(_ []string) error {
 		return errors.New("Cannot use --nvme-health and --nvme-meta together")
 	}
 
-	ctx := context.Background()
 	req := &control.StorageScanReq{NvmeHealth: cmd.NvmeHealth, NvmeMeta: cmd.NvmeMeta}
 	req.SetHostList(cmd.hostlist)
-	resp, err := control.StorageScan(ctx, cmd.ctlInvoker, req)
+	resp, err := control.StorageScan(context.Background(), cmd.ctlInvoker, req)
+	if err != nil {
+		return err
+	}
 
 	if cmd.jsonOutputEnabled() {
 		if cmd.Verbose {
 			cmd.log.Error("--verbose flag ignored if --json specified")
 		}
 
-		return cmd.outputJSON(resp, err)
+		return cmd.outputJSON(resp, resp.Errors())
 	}
 
-	if err != nil {
+	var outErr strings.Builder
+	if err := pretty.PrintResponseErrors(resp, &outErr); err != nil {
 		return err
 	}
-
-	var bld strings.Builder
-	verbose := pretty.PrintWithVerboseOutput(cmd.Verbose)
-	if err := pretty.PrintResponseErrors(resp, &bld); err != nil {
-		return err
+	if outErr.Len() > 0 {
+		cmd.log.Error(outErr.String())
 	}
-	if cmd.NvmeHealth {
+
+	var out strings.Builder
+	switch {
+	case cmd.NvmeHealth:
 		if cmd.Verbose {
 			cmd.log.Info("--verbose flag ignored if --nvme-health specified")
 		}
```

Comment:
Likewise these messages don't really help anyone IMO. They don't need to be removed in this PR, but should go at some point.